ocw_data_parser/ocw_data_parser_test.py

```diff
@@ -2,29 +2,189 @@
 Tests for ocw_data_parser
 """
 import json
-import unittest
+import pytest
+import os
+import shutil
+import responses
+import boto3
+from moto import mock_s3
+from tempfile import TemporaryDirectory
+from ocw_data_parser.ocw_data_parser import CustomHTMLParser, OCWParser
+import ocw_data_parser.test_constants as constants
 
-from ocw_data_parser import OCWParser
+"""
+Tests for OCW Parser
+"""
+
+@pytest.fixture(autouse=True, scope="session")
+def s3_bucket():
+    with mock_s3():
+        conn = boto3.client("s3",
+                            aws_access_key_id="testing",
+                            aws_secret_access_key="testing")
+        conn.create_bucket(Bucket="testing")
+        responses.add_passthru("https://")
+        responses.add_passthru("http://")
+        s3 = boto3.resource("s3",
+                            aws_access_key_id="testing",
+                            aws_secret_access_key="testing")
+        s3_bucket = s3.Bucket(name="testing")
+        yield s3_bucket
+
+@pytest.fixture(autouse=True, scope="function")
+def ocw_parser():
+    """
+    Instantiate an OCWParser object and run functions depending on args passed in
+    """
+    with TemporaryDirectory() as destination_dir:
+        parser = OCWParser(course_dir=constants.COURSE_DIR,
+                            destination_dir=destination_dir,
+                            static_prefix=constants.STATIC_PREFIX)
+        yield parser
+
+@pytest.fixture(autouse=True, scope="session")
+def ocw_parser_s3():
+    with TemporaryDirectory() as destination_dir:
+        parser = OCWParser(course_dir=constants.COURSE_DIR,
+                            destination_dir=destination_dir,
+                            static_prefix=constants.STATIC_PREFIX)
+        parser.setup_s3_uploading(
+            s3_bucket_name="testing",
+            s3_bucket_access_key="testing",
+            s3_bucket_secret_access_key="testing",
+            folder="testing"
+        )
+        yield parser
+
+@pytest.fixture(autouse=True)
+def course_id(ocw_parser):
+    yield ocw_parser.master_json["short_url"]
+
+def test_no_params(ocw_parser):
+    """
+    Test that an OCWParser with no params raises an exception
+    """
+    with pytest.raises(Exception):
+        OCWParser()
+
+def test_html_parser_output_list(ocw_parser):
+    """
+    Test passing in an output_list to the CustomHTMLParser
+    """
+    output_list = ["test"]
+    parser = CustomHTMLParser(output_list=output_list)
+    assert "test" == parser.output_list[0]
+
+def test_parser_loaded_jsons(ocw_parser):
+    """
+    Test instantiating a parser with preloaded jsons
+    """
+    try:
+        assert OCWParser(loaded_jsons=ocw_parser.jsons)
+    except:
+        fail("instantiating parser with preloaded jsons failed")
```

Comment:
These four lines can be rewritten as:

    assert OCWParser(loaded_jsons=ocw_parser.jsons), "instantiating parser with preloaded jsons failed"

Can you make a similar change for the other `fail(...)` calls in this file and `utils_test.py`?

Reply:
Also, feel free to take or leave the assert message. We generally don't use them in our projects, but include them if you think it adds context.

Reply:
And finally, `fail` does not appear to be a valid function